### PR TITLE
Add the upgrade hooks to step cucushift-upgrade-rhel-worker 

### PR DESCRIPTION
**Pull request by gpei**
Add the support for testing the upgrade hooks to Prow step.

### CODE REVIEW

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-4.18-upgrade-from-stable-4.17-azure-ipi-proxy-workers-rhcos-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-4.18-upgrade-from-stable-4.17-azure-ipi-proxy-workers-rhcos-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-4.18-upgrade-from-stable-4.17-azure-ipi-proxy-workers-rhcos-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-4.17-upgrade-from-stable-4.16-azure-ipi-proxy-workers-rhcos-rhel8-f28

**Comment by gpei** [comment]
The 4.17 to 4.18 job failed for a bug, fixing it in the openshift-ansible repo: https://github.com/openshift/openshift-ansible/pull/12523

File: ci-operator/step-registry/cucushift/upgrade/rhel-worker/cucushift-upgrade-rhel-worker-commands.sh [hunk 1]
```diff
@@ -242,6 +300,7 @@ echo -e "The source release version is gotten from clusterversion resource, that
 if [[ $(oc get nodes -l node.openshift.io/os_id=rhel) != "" ]]; then
     run_command "oc get node -owide"
     rhel_repo
+    rhel_pre_upgrade
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/upgrade/rhel-worker/cucushift-upgrade-rhel-worker-commands.sh:303]: I guess we also can apply the step to `openshift-upgrade-qe-sanity-disconnected`, `openshift-upgrade-qe-sanity` and `openshift-upgrade-qe-test-disconnected`

Comment (gpei) [reply ci-operator/step-registry/cucushift/upgrade/rhel-worker/cucushift-upgrade-rhel-worker-commands.sh:303]: @jianlinliu I see the step `cucushift-upgrade-rhel-worker` is already included in the above three chains? 
>openshift-upgrade-qe-sanity-disconnected, openshift-upgrade-qe-sanity and openshift-upgrade-qe-test-disconnected

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/upgrade/rhel-worker/cucushift-upgrade-rhel-worker-commands.sh:303]: Ah, my fault, I mis-read the chains.

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-4.17-upgrade-from-stable-4.16-azure-ipi-proxy-workers-rhcos-rhel8-f28

**Comment by gpei** [comment]
In the latest 4.16 to 4.17 upgrade [job](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58434/rehearse-58434-periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-4.17-upgrade-from-stable-4.16-azure-ipi-proxy-workers-rhcos-rhel8-f28/1853982262731739136/artifacts/azure-ipi-proxy-workers-rhcos-rhel8-f28/cucushift-upgrade-rhel-worker/build-log.txt), the `cucushift-upgrade-rhel-worker` step passed.
```
INFO[2024-11-06T07:16:08Z] Running step azure-ipi-proxy-workers-rhcos-rhel8-f28-cucushift-upgrade-rhel-worker. 
INFO[2024-11-06T07:39:18Z] Step azure-ipi-proxy-workers-rhcos-rhel8-f28-cucushift-upgrade-rhel-worker succeeded after 23m10s. 
```

```
Check the upgrade hook flags created
The hook pre_cordon.yaml was executed.
The hook pre_uncordon.yaml was executed.
The hook pre_upgrade.yaml was executed.
The hook post_upgrade.yaml was executed.
```

**Comment by gpei** [comment]
@jianlinliu please have another check

**Pull request merged**